src/Core/PerformanceMonitor.cs

```diff
@@ -13,7 +13,11 @@ namespace Microsoft.Quantum.IQSharp
     public class DeviceCapabilitiesEvent : Event<DeviceCapabilitiesArgs>
     {}
 
+    #if NET5_0 || NET6_0 || NET5_0_OR_GREATER
     public record DeviceCapabilitiesArgs
```

Comment:
I wonder if we should postpone moving to records until we require .NET 5.0 or greater for the whole project.
Otherwise we will have to spread the code with compiler conditionals like you suggested and not sure it's worth the effort...

Reply:
Agreed, only did so in this case because we're as close to November's release of .NET 6, such that we should be able to remove this soon.